### PR TITLE
Added /who command

### PR DESCRIPTION
**Pull request by ruddles**
Added /who command:

/who - lists all users
/who [name] - lists rooms the user is in.

### CODE REVIEW

File: Chat/Chat.cs [hunk 1]
```diff
@@ -438,6 +443,36 @@ namespace SignalR.Samples.Hubs.Chat
...
+
+            var exactUserMatch = _db.Users.FirstOrDefault(s => s.Name.Equals(name, StringComparison.OrdinalIgnoreCase));
+
+            if (exactUserMatch != null)
+            {
+                Caller.showUsersRoomList(exactUserMatch.Name, exactUserMatch.Rooms.Select(r => r.Name));
+                return;
+            }
+
+            var users = _db.Users.Where(s => s.Name.IndexOf(name, StringComparison.OrdinalIgnoreCase) != -1);
```
Comment (davidfowl) [inline Chat/Chat.cs:464]: Why IndexOf and note Equals? Ah it's a search

File: Chat/Chat.cs [hunk 2]
```diff
@@ -438,6 +443,36 @@ namespace SignalR.Samples.Hubs.Chat
...
+
+            if (users.Count() == 1)
+            {
+                Caller.showUsersRoomList(users.First().Name, users.First().Rooms.Select(r => r.Name));
```
Comment (davidfowl) [inline Chat/Chat.cs]: Change this to:

```
var user = users.First();
Caller.showUsersRoomList(user.Name, user.Rooms.Select(r => r.Name));
```

File: Chat/Chat.cs (hunk 1 above)
Comment (ruddles) [inline Chat/Chat.cs:464]: Aye, allows partial, case-insensitive, matches

File: Chat/Chat.cs (hunk 2 above)
Comment (ruddles) [inline Chat/Chat.cs]: Done

**Pull request merged**